c/src/lib/libbsp/arm/shared/stm32fxxxx/i2c/hal-i2c.c

```diff
@@ -0,0 +1,331 @@
+
+#include <hal-i2c.h>
+#include <hal-i2c-init.h>
+
+#include <math.h>
+#include <rtems/irq-extension.h>
+
+stm32_i2c_bus *bus[MAX_I2C_INSTANCES];
+I2C_HandleTypeDef I2cHandle;
+
+stm32_i2c_pin_config i2c_pin_config[MAX_I2C_INSTANCES] = {
+#if (STM32_ENABLE_I2C1)
+													{.port 		= I2C1_PORT,
+													.sck_pin 	= I2C1_SCK_PIN,
+													.sda_pin 	= I2C1_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C1},
+#endif
+#if (STM32_ENABLE_I2C2)
+													{.port 		= I2C2_PORT,
+													.sck_pin 	= I2C2_SCK_PIN,
+													.sda_pin 	= I2C2_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C2},
+#endif
+#if (STM32_ENABLE_I2C3)
+													{.port 		= I2C3_PORT,
+													.sck_pin 	= I2C3_SCK_PIN,
+													.sda_pin 	= I2C3_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C3},
+#endif
+#if (STM32_ENABLE_I2C4)
+													{.port 		= I2C4_PORT,
+													.sck_pin 	= I2C4_SCK_PIN,
+													.sda_pin 	= I2C4_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C4},
+#endif
+};
+
+const bool I2C_Enable[MAX_I2C_INSTANCES] = {
+#ifndef STM32_ENABLE_I2C1
+		false,
+#else
+		STM32_ENABLE_I2C1,
+#endif
+#ifndef STM32_ENABLE_I2C2
+		false,
+#else
+		STM32_ENABLE_I2C2,
+#endif
+#ifndef STM32_ENABLE_I2C3
+		false,
+#else
+		STM32_ENABLE_I2C3,
+#endif
+#ifndef STM32_ENABLE_I2C4
+		false,
+#else
+		STM32_ENABLE_I2C4,
+#endif
+};
+
+/**
+ * I2C Own Device Addresses
+ */
+uint32_t I2C_Address[MAX_I2C_INSTANCES] = {
+#if (STM32_ENABLE_I2C1)
+		STM32_I2C1_ADDRESS,
+#else
+		0x00,
+#endif
+#if (STM32_ENABLE_I2C2)
+		STM32_I2C2_ADDRESS,
+#else
+		0x00,
+#endif
+#if (STM32_ENABLE_I2C3)
+		STM32_I2C3_ADDRESS,
+#else
+		0x00,
+#endif
+#if (STM32_ENABLE_I2C4)
+		STM32_I2C4_ADDRESS,
+#else
+		0x00,
+#endif
+};
+
+/*********** Private Funvtions **********************/
+
+
+/**
+ * I2C get instance based vector number
+ */
+IRQn_Type stm32_get_i2c_vec_num(I2C_Instance i2c_instance)
+{
+	switch(i2c_instance)
+	{
+	case I2C_ONE:	return (IRQn_Type)GET_I2C_VEC((I2C_ONE++));
+					break;
+	case I2C_TWO:	return (IRQn_Type)GET_I2C_VEC(I2C_TWO++);
+					break;
+	case I2C_THREE:	return (IRQn_Type)GET_I2C_VEC(I2C_THREE++);
+					break;
+	case I2C_FOUR:	return (IRQn_Type)GET_I2C_VEC(I2C_FOUR++);
+					break;
+	}
+}
+
+/**
+ * Initialize I2C clock
+ */
+void stm32_i2c_initialize_i2c_clock(I2C_Instance i2c_instance)
+{
+	switch(i2c_instance)
+	{
+	case I2C_ONE:	__HAL_RCC_I2C1_CLK_ENABLE();
+					break;
+	case I2C_TWO:	__HAL_RCC_I2C2_CLK_ENABLE();
+					break;
+	case I2C_THREE:	__HAL_RCC_I2C3_CLK_ENABLE();
+					break;
+#if (STM32_ENABLE_I2C4)
+	case I2C_FOUR:	__HAL_RCC_I2C4_CLK_ENABLE();
+					break;
+#endif
+	}
+}
+
+/**
+ * Returns the base address of the I2C instances
+ */
+I2C_TypeDef* stm32_i2c_get_i2c_instance(I2C_Instance i2c_instance)
+{
+	switch(i2c_instance)
+	{
+	case I2C_ONE:	return I2C1;
+					break;
+	case I2C_TWO:	return I2C2;
+					break;
+	case I2C_THREE:	return I2C3;
+					break;
+#if (STM32_ENABLE_I2C4)
```

Comment:
Why is only this case conditionally compiled based upon the enabling configuration state?